CMakeLists.txt

```diff
@@ -687,11 +750,18 @@ ExternalProject_Add (libarchive
     -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
     -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
     $<$<BOOL:${CMAKE_VERBOSE_MAKEFILE}>:-DCMAKE_VERBOSE_MAKEFILE=ON>
-    -DCMAKE_PREFIX_PATH=${CMAKE_BINARY_DIR}/lz4
     -DCMAKE_DEBUG_POSTFIX=_d
-    -DCMAKE_BUILD_TYPE=$<IF:$<STREQUAL:${CMAKE_BUILD_TYPE},Debug>,Debug,Release>
+    $<$<NOT:$<BOOL:${is_multiconfig}>>:-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}>
     -DENABLE_LZ4=ON
     -ULZ4_*
+    -DLZ4_INCLUDE_DIR=$<TARGET_PROPERTY:lz4_lib,INTERFACE_INCLUDE_DIRECTORIES>
+    # because we are building a static lib, this lz4 library doesn't
+    # actually matter since you can't generally link static libs to other static
+    # libs. The include files are needed, but the library itself is not (until
+    # we link our application, at which point we use the lz4 we built above).
+    # nonetheless, we need to provide a library to libarchive else it will
+    # NOT include lz4 support when configuring
+    -DLZ4_LIBRARY=$<IF:$<CONFIG:Debug>,$<TARGET_PROPERTY:lz4_lib,IMPORTED_LOCATION_DEBUG>,$<TARGET_PROPERTY:lz4_lib,IMPORTED_LOCATION_RELEASE>>
```

Comment:
genex's are a real pain to format nicely in CMake, because extra whitespace gets interpreted in various ways depending on the type of argument. This is big CMake flaw. That said, the only other way that I could find to shorten this line is like this:

```
    $<$<CONFIG:Debug>:
      "-DLZ4_LIBRARY=$<TARGET_PROPERTY:lz4_lib,IMPORTED_LOCATION_DEBUG>"
    >
    $<$<NOT:$<CONFIG:Debug>>:
      "-DLZ4_LIBRARY=$<TARGET_PROPERTY:lz4_lib,IMPORTED_LOCATION_RELEASE>"
    >
```
If you think that's a net improvement, I'm happy to change it. That said, I think the existing way is slightly easier to comprehend (its intent), but has the disadvantage of being very long.

Reply:
I agree the long line is a bit clearer. I mean, it's kinda gibberish either way, but less gibberish with the long line. :grinning: :smirk: